dataproc/src/main/java/DeleteCluster.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START dataproc_delete_cluster]
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.dataproc.v1.ClusterControllerClient;
+import com.google.cloud.dataproc.v1.ClusterControllerSettings;
+import com.google.cloud.dataproc.v1.ClusterOperationMetadata;
+import com.google.protobuf.Empty;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public class DeleteCluster {
+
+  public static void deleteCluster() throws IOException, InterruptedException, ExecutionException {
```

Comment:
I think I may have told you the wrong thing when I said let `ExecutionException` bubble up (I was mixing it up with `InterruptedException`). 

It looks like `ExecutionException` is being used to wrap errors returned by the API - so I'm not sure if this should be caught or not. I'm guessing it might be best to show the user how to unwrap something like the following:

```java
} catch (ExecutionException ex) {
     System.out.println("The service returned an error to you request: ", ex.getCause());
}
```

@lesv - thoughts? raise or catch?